lighthouse-core/audits/csp-xss.js

```diff
@@ -155,17 +127,19 @@ class CspXss extends Audit {
     }
 
     // TODO: Add severity icons for bypasses and warnings.
-    const bypasses = this.collectBypassResults(rawCsps);
-    const warnings = this.collectWarningResults(rawCsps);
-    const syntax = this.collectSyntaxResults(rawCsps);
+    const {bypasses, warnings, syntax} = evaluateRawCspsForXss(rawCsps);
+
+    const results = [
+      ...this.constructSyntaxResults(syntax, rawCsps),
+      ...bypasses.map(this.findingToTableItem),
+      ...warnings.map(this.findingToTableItem),
+    ];
 
     // Add extra warning for a CSP defined in a meta tag.
     if (cspMetaTags.length) {
-      warnings.push({description: str_(UIStrings.metaTagMessage), directive: undefined});
+      results.push({description: str_(UIStrings.metaTagMessage), directive: undefined});
     }
 
-    const results = [...syntax, ...bypasses, ...warnings];
-
     /** @type {LH.Audit.Details.Table['headings']} */
     const headings = [
       /* eslint-disable max-len */
```

Comment:
when looking at the results, it felt like `directive` column first would be better. as each description is useful only after knowing what directive you're talking about.

that said, we don't often do the short column on the left, so it may look weird.

so... wdyt?

Reply:
I remember thinking it looked weird, but that may change with the severity icons. I'm thinking we move this comment to https://github.com/GoogleChrome/lighthouse/pull/12240.

Reply:
sure. also iirc the css has the last column as `text-align:right`, which would definitely make `description`s look odd. :)